### DISCUSSION TITLE
Make Fails, Minimal Error Message

### DISCUSSION THREAD

**Discussion by d-v-t** [idaholab/moose#24442] [Q&A Getting Started]
## Bug Description
When I run make on my Ubuntu 22.04.2 machine in the test directory as described [here](https://mooseframework.inl.gov/getting_started/installation/conda.html), I get a bunch of file not found errors for moose/libmesh/installed. I suspect this isn't the main error as moose/libmesh is completely empty.

Make eventually spits out the following error:

`make: *** [/path_to_moose/moose/framework/build.mk:149: /path_to_moose/moose/framework/build/unity_src/src_Unity..opt.lo] Error 127`

Poking around in the .mk file lead me to believe that it would take far too long to fix whatever was causing this error. I don't have any experience in C, so I apologize for any lack of knowledge in advance. I've had a terrible time using make files no matter who the distributor, so it's likely I'm making a very common mistake.

## Steps to Reproduce
-Install moose as described by install guide (except instead of ~/projects I used my own directory)
-Run make -j 6

## Impact
I can't install moose.

**Comment by milljm** [comment]
Can we know the directory you are working from? Is it somewhere in your home directory (or somewhere where you always have write permissions without the use of `sudo`)?

File not found errors during `make` might be that the Conda environment didn't properly activate. The error, I am guessing was something like this?

```pre
❯ make
/bin/sh: /Users/milljm/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/milljm/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/milljm/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/milljm/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/milljm/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/milljm/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/milljm/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/milljm/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
/bin/sh: /Users/milljm/projects/moose/libmesh/installed/contrib/bin/libmesh-config: No such file or directory
<trimmed>


❯ echo $LIBMESH_DIR
<nothing returns>
```

This occurs due to our Make system trying to utilize either `$LIBMESH_DIR` (which as I demonstrate above does not exist), or use moose/libmesh instead. Which also does not exist when following our Conda instructions (because our Conda instructions eventually lead you to installing a pre-built version of libMesh via `moose-dev`.

The fix (for my demonstration above), is to activate the `moose` environment:

```bash
❯ mamba activate moose
❯ echo $LIBMESH_DIR
/Users/milljm/mambaforge3/envs/moose/libmesh
```

I now have LIBMESH_DIR properly available, and so a `make` will proceed normally.
